src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6181,6 +6204,72 @@ IOperation popTarget(IOperation originalTarget)
                         throw ExceptionUtilities.UnexpectedValue(originalTarget.Kind);
                 }
             }
+
+            static bool onlyContainsEmptyLeafNestedInitializers(IMemberInitializerOperation memberInitializer)
+            {
```

Comment:
>{ [](http://example.com/codeflow?start=12&length=1)

Consider asserting that the shape of ```memberInitializer.InitializedMember``` is going to be handled by the ```addIndexes``` helper. We may never end up calling it for existing test scenarios because initializers aren't empty, but we should be prepared to handle them.

Reply:
I'm not sure I follow. We can't assert with a call to addIndexes because that has side-effects. Are you suggesting to have `addIndexes` return all the indexes that were found?

Reply:
> I'm not sure I follow. We can't assert with a call to addIndexes because that has side-effects. Are you suggesting to have addIndexes return all the indexes that were found?

```addIndexes``` can handle ```IPropertyReferenceOperation```,  ```IImplicitIndexerReferenceOperation```,  ```IArrayElementReferenceOperation```, ```IDynamicIndexerAccessOperation```, ```FieldReferenceOperation``` and ```NoneOperation``` as ```memberInitializer.InitializedMember```. I am suggesting to assert that ```memberInitializer.InitializedMember``` doesn't have any other shape at this point.

Reply:
Note, it looks like ```tryPushTarget``` is prepared to handle more shapes, and, at the same time, it is more lenient. Meaning, it doesn't crash on something unexpected. Therefore, consider if it would make sense to take a similar approach here. Specifically, have a check for shapes (not just an assert) that we are prepared to handle in ```addIndexes```. If an unexpected shape comes through, assert and return false.